query-parser/src/parse.rs

```diff
@@ -172,37 +207,142 @@ impl<I> FindSp<I>
 //     `?x .      `     = FindScalar
 //     `[?x ?y ?z]`     = FindTuple
 //
+#[allow(dead_code)]
 pub fn find_seq_to_find_spec(find: &[edn::Value]) -> FindParseResult {
-    FindSp::find()
+    Find::find()
         .parse(find)
         .map(|x| x.0)
         .map_err(|_| FindParseError::Err)
 }
 
+#[allow(dead_code)]
+pub fn clause_seq_to_patterns(clauses: &[edn::Value]) -> WhereParseResult {
+    Where::clauses()
+        .parse(clauses)
+        .map(|x| x.0)
+        .map_err(|_| WhereParseError::Err)
+}
+
 #[cfg(test)]
 mod test {
     extern crate combine;
     extern crate edn;
     extern crate mentat_query;
+    extern crate ordered_float;
 
     use self::combine::Parser;
-    use self::mentat_query::{Element, FindSpec, Variable};
+    use self::ordered_float::OrderedFloat;
+    use self::mentat_query::{
+        Element,
+        FindSpec,
+        NonIntegerConstant,
+        Pattern,
+        PatternNonValuePlace,
+        PatternValuePlace,
+        SrcVar,
+        Variable,
+    };
 
     use super::*;
 
     #[test]
+    fn test_pattern_mixed() {
+        let e = edn::PlainSymbol::new("_");
```

Comment:
I think it's worth parsing EDN from strings for these tests.  It's damn hard to make sense of the input.  Next patch?